jersey-servers/src/main/java/com/palantir/remoting2/servers/jersey/DeflateFilter.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting2.servers.jersey;
+
+import com.google.common.net.HttpHeaders;
+import com.jcraft.jzlib.DeflaterOutputStream;
+import java.io.IOException;
+import java.util.List;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+
+public final class DeflateFilter implements ContainerResponseFilter {
+
+    private static final String DEFLATE = "deflate";
+
+    @Override
+    public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext)
+            throws IOException {
+        if (acceptsDeflateEncoding(requestContext) && !alreadyEncoded(responseContext)) {
+            responseContext.getHeaders().add(HttpHeaders.CONTENT_ENCODING, DEFLATE);
+            responseContext.setEntityStream(new DeflaterOutputStream(responseContext.getEntityStream()));
```

Comment:
yep added it.

explanation: https://blog.stackpath.com/accept-encoding-vary-important